src/docfx/lib/schema/JsonSchemaValidator.cs

```diff
@@ -381,31 +381,88 @@ private void ValidateDeprecated(JsonSchema schema, string name, JToken token, Li
+            {
+                var (fieldName, fieldIndex) = GetFieldNameAndIndex(fieldNameWithIndex);
+                if (map.TryGetValue(fieldName, out var fieldRawValue))
                 {
-                    foreach (var (dependentFieldName, allowLists) in enumDependencyRules)
+                    var fieldValue = GetFieldValue(fieldRawValue, fieldIndex);
+
+                    if (fieldValue == null)
+                    {
+                        return;
+                    }
+
+                    if (allowList.TryGetValue(fieldValue, out var nextEnumDependencies))
                     {
-                        if (map.TryGetValue(dependentFieldName, out var dependentFieldValue))
+                        ValidateEnumDependencies(nextEnumDependencies, fieldNameWithIndex, fieldName, fieldRawValue, fieldValue, map, errors);
+                    }
+                    else
+                    {
+                        if (string.IsNullOrEmpty(dependentFieldNameWithIndex))
                         {
-                            if (allowLists.TryGetValue(dependentFieldValue, out var allowList) &&
-                                Array.IndexOf(allowList, fieldValue) == -1)
-                            {
-                                errors.Add((dependentFieldName, Errors.InvalidPairedAttribute(JsonUtility.GetSourceInfo(map), fieldName, fieldValue, dependentFieldName, dependentFieldValue)));
-                            }
+                            errors.Add((fieldName, Errors.InvalidValue(
+                                JsonUtility.GetSourceInfo(fieldValue),
+                                fieldRawValue.Type == JTokenType.Array ? fieldNameWithIndex : fieldName,
+                                fieldValue)));
                         }
                         else
                         {
-                            errors.Add((fieldName, Errors.MissingPairedAttribute(JsonUtility.GetSourceInfo(map), fieldName, dependentFieldName)));
+                            errors.Add((dependentFieldName, Errors.InvalidPairedAttribute(
+                                JsonUtility.GetSourceInfo(fieldValue),
+                                fieldRawValue.Type == JTokenType.Array ? fieldNameWithIndex : fieldName,
+                                fieldValue,
+                                dependentFieldRawValue.Type == JTokenType.Array ? dependentFieldNameWithIndex : dependentFieldName,
+                                dependentFieldValue)));
                         }
                     }
                 }
             }
         }
 
+        // For string type: name = 'topic', output = ('topic', 0) or name = 'topic[0]', output = ('topic', 0)
+        // For array type: name = 'topic[0]', output = ('topic', 0) or name = 'topic[1]', output = ('topic', 1) or ...
+        private (string, int) GetFieldNameAndIndex(string name)
+        {
+            var match = Regex.Match(name, @"\[\d+\]$");
```

Comment:
check if name contains `[` and `]` before using `Regex` because `Regex is potentially slow

Reply:
make sense